app/controllers/project1.php

```diff
@@ -0,0 +1,279 @@
+            return Redirect::back()->withErrors($validator)->withInput();
+        }
+
+		$project->name 			= Input::get('name');
+		$project->github 		= Input::get('github');
+		$project->production 	= Input::get('production');
+		$project->stage 		= Input::get('stage');
+		$project->dev 			= Input::get('dev');
+		$project->save();
+
+		return Redirect::back()->with('success', "The project " .Input::get('name'). " has been updated.");
+	}
+
+	/**
+	 * Remove the specified resource from storage.
+	 * DELETE /projects/{id}
+	 *
+	 * @param  int  $id
+	 * @return Response
+	 */
+	public function destroy($id)
+	{
+		$pTitle		=	"Projects";
+
+		$project 	= 	Project::find(Input::get("id"));
+
+		// delete everything associated with project
+		Task::where('project_id',Input::get("id"))->delete();		
+		Credential::where('project_id',Input::get("id"))->delete();
+		$project->members()->detach();
+
+
+		// delete the project
+		$project->delete();
+		
+		$counter 	=	0;
+		$user 		=	User::find(Auth::id());
+		$projects 	=	$user->projects()->get();
+		$inProjects =  $user->inProjects()->orderBy('created_at', 'desc')->take(5)->get();		
+								
+		return View::make('projects.index',compact(['projects','counter','inProjects','pTitle']));		
+	}
+
+	/**
+	 * Invites a user to the given project.
+	 * @return Redirect
+	 */
+	public function invite($id){
+		// Validation
+		$rules = ['email' => 'required|email|exists:users,email'];
+		$messages = [ 'exists' => 'That email is not currently associated with a user.',];
+
+		$validator = Validator::make(Input::all(), $rules, $messages);
+
+        if ($validator->fails())
+		{
+			return Redirect::back()->withErrors($validator)->withInput();
+		}
+
+		// Get the id of the user being sent the invite
+		$user_id = DB::table('users')->whereEmail(Input::get('email'))->pluck('id');
+
+		if( count(Projectuser::whereUserId($user_id)->whereProjectId($id)->get()) != 0 )
+		{	
+			$validator->getMessageBag()->add('email', 'A user with that email has already been invited.');
+			return Redirect::back()->withErrors($validator)->withInput();
+		}
+
+		// Save the relationship between user and project.
+		$pu				= new Projectuser;
+		$pu->project_id	=	$id;
+		$pu->user_id	=	$user_id;
+		$pu->save();
+
+		// Prepare email invitation & send it
+		$project_name	= Project::find($id)->pluck('name');
+		$project_url 	= url() . '/projects/'.$id;
+		sendProjectInviteMail(Input::get('email'), $project_name, $project_url);
+
+		return Redirect::back()->with('success', "A new member has been added to this project.");;
+	}
+
+	/**
+	 * @param int $id
+	 *
+	 * Removes a member from a given project
+	 * @return redirect
+	 */
+	public function remove($id){
+		$project = Project::find($id);
+		$project->members()->detach(Input::get('member_id'));
+
+		return Redirect::back();
+	}
+
+	public function credentials($id){
+		$project 		=	Project::find($id);
+
+		$total_weight	=	$project->tasks()->where('state','incomplete')->sum('weight');
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this unused "$total_weight" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://172.27.59.137:9000/coding_rules#rule_key=php%3AS1481)